Feat/pages/company register and 404

gooddinosaur opened the pull request:
## Checklist
- [x] 404 page (Name the file `not-found.tsx` and place it under `src/app` for automatic setup.)
- [x] Company Register Page

## Screenshots / Demos
- 404 page
<img width="1919" height="923" alt="image" src="https://github.com/user-attachments/assets/43cfccd3-fc5d-449a-a411-7bc639462d66" />


- Company Register Page

<img width="1900" height="925" alt="image" src="https://github.com/user-attachments/assets/9d97db78-156e-42be-aa98-9251e6526c56" />
<img width="1888" height="522" alt="image" src="https://github.com/user-attachments/assets/0d3e072a-1e2f-43f3-99a0-5578ac21379a" />

- Company Register Page (Submit with empty fields)

<img width="711" height="746" alt="image" src="https://github.com/user-attachments/assets/60d7e058-79f0-4204-87b0-e23fa97cb2df" />

- Company Register Page (Upload files)

<img width="692" height="597" alt="image" src="https://github.com/user-attachments/assets/e57b9115-c7b0-4763-af3d-7bbf655219c1" />

gooddinosaur committed:
feat(page): Add 404 page

gooddinosaur committed:
feat(company): Add company register page

gooddinosaur committed:
chore(color): change color of some texts in company register

gooddinosaur committed:
fix(ui): make the company register page look better

gooddinosaur committed:
feat(ui): Link confirm modal to submit button

gooddinosaur committed:
fix(validation): Fix logic to validate email,  based on recommendation of SonarQube

gooddinosaur committed:
fix(validation): fix email field can input thai language email

catwithasword reviewed (changes requested):
1. /app should be layout-focused so it shouldnt be actions.ts in that location. you can struct files like 
```
src/features/company-register/
├─ components/
│  └─ CompanyRegisterForm.tsx        # UI / client component (or wrapper)
├─ hooks/
│  └─ useCompanyRegister.ts         # client hook for submission state, modal, etc.
├─ server/
│  └─ actions.server.ts             # "use server" registerCompany (server action / service)
├─ services/
│  └─ upload.ts  
```
2. No need for adding Navbar and Footer in your file cuz it will be overlapped with the layout

gooddinosaur committed:
fix(ui): Remove Navbar and Footer from page.tsx in company-register dir

gooddinosaur committed:
fix(file): Restructure files and add index.ts in src/features/company-register

gooddinosaur commented:
### Changes
- Restructured files based on feedback
- Removed `Navbar` and `Footer` from `company-register/page.tsx`
- Renamed `actions.ts` → `actions.server.ts`

catwithasword commented:
1. Can you provide a preview image when it was uploaded?
2. I think you forgot ',' between PNg and PDF
<img width="927" height="382" alt="image" src="https://github.com/user-attachments/assets/4924a006-2fd0-4d57-94ff-4983b81b27e7" />

catwithasword commented:
3. hover effact in dropdown would be great. on small screen i dont know what my mouse is pointing to.

gooddinosaur committed:
feat(company): Add preview image when it was uploaded

gooddinosaur committed:
chore(form): Change hover color of dropdown of Industry and Company Size

gooddinosaur commented:
### Changes
- Add preview image when uploaded
<img width="699" height="337" alt="image" src="https://github.com/user-attachments/assets/51e5da72-f5a0-4ef8-8ce5-89eb0c57fff0" />

- Add `,` between PNG and PDF
<img width="282" height="108" alt="image" src="https://github.com/user-attachments/assets/8d421aeb-b6ff-478b-8cba-311935e55ba5" />

- Add hover in dropdown of Industry and Company Size
<img width="359" height="315" alt="image" src="https://github.com/user-attachments/assets/fa6368d9-2f8d-4e4a-ab03-c7a1fff723a8" />
<img width="336" height="270" alt="image" src="https://github.com/user-attachments/assets/1c0b78f4-0ab4-49ec-817d-a0b7b3480480" />

gooddinosaur committed:
chore(company): Add "Uploaded: " to preview image card

catwithasword reviewed (approved):
ok. good work bro. you deserve an approve
